### DISCUSSION TITLE
security and ACK

### DISCUSSION THREAD

**Discussion by pipiche38** [zigpy/zigpy#753] [Q&A] [answered]
My question is related to the zigpy to radio module lib and in this particular case zigpy_zigate

While doing my test, I observe that the RAW commands are always used with security set to 0, while in the NXP documentation

> eSecurityMode is the security mode for the data transfer, one of:
>         0x00 : ZPS_E_APL_AF_UNSECURE (no security enabled)
>         0x01 : ZPS_E_APL_AF_SECURE Application-level security using link key and network key)
>         0x02 : ZPS_E_APL_AF_SECURE_NWK (Network-level security using network key)
>         0x10 : ZPS_E_APL_AF_SECURE | ZPS_E_APL_AF_EXT_NONCE (Application-level security using link key and network key with the extended NONCE included in the frame)
>         0x20 : ZPS_E_APL_AF_WILD_PROFILE (May be combined with above flags using OR operator. Sends the message using the wild card profile (0xFFFF) instead of the profile in the associated Simple descriptor)

Is that security mode managed by zigpy , or does it needs to be managed by the Radio lib ?

I also see that all raw_aps_data_request are send with ACK, is there any mecanism on the zigpy level to decide if ACK is required or not ? when sending the RAW command ?

**Comment by Adminiuga** [comment, marked as answer]
Currently zigpy itself does not manage the security settings and leaves it to the radio lib. AFAIK ATM all radio libs use NWK level security.

Zigpy also does not mandate/controls the use of APS ACK, but I'm thinking at least allowing disabling the default response from the application layer.
